### PR TITLE
[RLlib] Fix tf timeline in ray local mode

### LINKED ISSUE #26511: [RLlib] AttributeError: module 'tensorflow' has no attribute 'RunOptions'
### What happened + What you expected to happen

# What happened
I ran PPO with the environment variable `"TF_TIMELINE_DIR"` to create the TensorFlow timeline for profiling. I got an error: 

```
ray.exceptions.RayTaskError(AttributeError): ray::PPO.train() (pid=3652791, ip=192.168.1.111, repr=PPO)
  File "/home/simon/git-projects/ray/.venv/lib/python3.9/site-packages/ray/tune/trainable/trainable.py", line 347, in train
    result = self.step()
  File "/home/simon/git-projects/ray/.venv/lib/python3.9/site-packages/ray/rllib/algorithms/algorithm.py", line 607, in step
    results, train_iter_ctx = self._run_one_training_iteration()
  File "/home/simon/git-projects/ray/.venv/lib/python3.9/site-packages/ray/rllib/algorithms/algorithm.py", line 2278, in _run_one_training_iteration
    self.try_recover_from_step_attempt(
  File "/home/simon/git-projects/ray/.venv/lib/python3.9/site-packages/ray/rllib/algorithms/algorithm.py", line 2093, in try_recover_from_step_attempt
    raise error
  File "/home/simon/git-projects/ray/.venv/lib/python3.9/site-packages/ray/rllib/algorithms/algorithm.py", line 2273, in _run_one_training_iteration
    results = self.training_step()
  File "/home/simon/git-projects/ray/.venv/lib/python3.9/site-packages/ray/rllib/algorithms/ppo/ppo.py", line 407, in training_step
    train_batch = synchronous_parallel_sample(
  File "/home/simon/git-projects/ray/.venv/lib/python3.9/site-packages/ray/rllib/execution/rollout_ops.py", line 96, in synchronous_parallel_sample
    sample_batches = [worker_set.local_worker().sample()]
  File "/home/simon/git-projects/ray/.venv/lib/python3.9/site-packages/ray/rllib/evaluation/rollout_worker.py", line 881, in sample
    batches = [self.input_reader.next()]
  File "/home/simon/git-projects/ray/.venv/lib/python3.9/site-packages/ray/rllib/evaluation/sampler.py", line 92, in next
    batches = [self.get_data()]
  File "/home/simon/git-projects/ray/.venv/lib/python3.9/site-packages/ray/rllib/evaluation/sampler.py", line 280, in get_data
    item = next(self._env_runner)
  File "/home/simon/git-projects/ray/.venv/lib/python3.9/site-packages/ray/rllib/evaluation/sampler.py", line 702, in _env_runner
    eval_results = _do_policy_eval(
  File "/home/simon/git-projects/ray/.venv/lib/python3.9/site-packages/ray/rllib/evaluation/sampler.py", line 1186, in _do_policy_eval
    eval_results[policy_id] = policy.compute_actions_from_input_dict(
  File "/home/simon/git-projects/ray/.venv/lib/python3.9/site-packages/ray/rllib/policy/tf_policy.py", line 313, in compute_actions_from_input_dict
    fetched = builder.get(to_fetch)
  File "/home/simon/git-projects/ray/.venv/lib/python3.9/site-packages/ray/rllib/utils/tf_run_builder.py", line 55, in get
    raise e
  File "/home/simon/git-projects/ray/.venv/lib/python3.9/site-packages/ray/rllib/utils/tf_run_builder.py", line 42, in get
    self._executed = _run_timeline(
  File "/home/simon/git-projects/ray/.venv/lib/python3.9/site-packages/ray/rllib/utils/tf_run_builder.py", line 76, in _run_timeline
    run_options = tf1.RunOptions(trace_level=tf.RunOptions.FULL_TRACE)
AttributeError: module 'tensorflow' has no attribute 'RunOptions'
```

# What you expected to happen
That the timeline gets recorded to the directory I provided. I saw that it needs `tf1` and not `tf` here. I will prepare a PR that fixes this. 

### Versions / Dependencies

Python 3.9.x
Ray dev3.0.0
Linux Fedora

### Reproduction script

```python
config = (
    ppo.PPOConfig()
    .environment(
        env="FrozenLake-v1",
    ).
    framework(
        framework="tf",
    )
    .training(
        num_sgd_iter=8,
    )
    .rollouts(
        num_rollout_workers=0,
    )
    .debugging(
        log_level="INFO"
    )

ray.init(ignore_reinit_error=True, local_mode=True)
os.mkdir("/tmp/tf_timeline_test")
os.environ["TF_TIMELINE_DIR"] = "/tmp/tf_timeline_test"

analysis = tune.run(
    "PPO",
    config=config.to_dict(),
    stop={
        "timesteps_total": 1
    },
    fail_fast=True 
) 
```

### Issue Severity

Low: It annoys or frustrates me.

### PR DESCRIPTION
**Pull request by ArturNiederfahrenhorst**
## Why are these changes needed?

Our tf1 imports make it so that tf.compat.v1.RunOptions is not available in ray's local mode.
I don't think we have to support this, rather we can default to warning user that the RunOptions flag `trace_level=tf.RunOptions.FULL_TRACE` will simply not be used, as the rest executes fine.

## Related issue number

Closes #26511

## Checks

- [x] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: rllib/utils/tf_run_builder.py [hunk 1]
```diff
@@ -73,7 +73,17 @@ def _run_timeline(sess, ops, debug_name, feed_dict=None, timeline_dir=None):
...
-        run_options = tf1.RunOptions(trace_level=tf.RunOptions.FULL_TRACE)
+        try:
+            run_options = tf1.RunOptions(trace_level=tf.RunOptions.FULL_TRACE)
+        except AttributeError:
+            run_options = None
+            # In local mode, tf1.RunOptions is not available, see #26511
+            if log_once("tf1.RunOptions_not_available"):
+                logger.exception(
+                    "Can not access tf.RunOptions.FULL_TRACE. This may be because "
+                    "you are using ray in local mode. RLlib will use timeline without "
```
Comment (sven1977) [inline rllib/utils/tf_run_builder.py:84]:
```suggestion
                    "you have used `ray.init(local_mode=True)`. RLlib will use timeline without "
```

**Review by sven1977 — APPROVED** [review]
Thanks for the fix @ArturNiederfahrenhorst , looks good to me (made one minor change to the comment). Let's wait for test to pass, then merge. :)

**Comment by ArturNiederfahrenhorst** [comment]
Thanks @sven1977 👍

File: rllib/utils/tf_run_builder.py [hunk 2]
```diff
@@ -73,7 +73,17 @@ def _run_timeline(sess, ops, debug_name, feed_dict=None, timeline_dir=None):
...
+        try:
+            run_options = tf1.RunOptions(trace_level=tf.RunOptions.FULL_TRACE)
+        except AttributeError:
+            run_options = None
+            # In local mode, tf1.RunOptions is not available, see #26511
+            if log_once("tf1.RunOptions_not_available"):
+                logger.exception(
+                    "Can not access tf.RunOptions.FULL_TRACE. This may be because "
+                    "you have used `ray.init(local_mode=True)`. RLlib will use timeline without "
+                    "`options=tf.RunOptions.FULL_TRACE`."
```
Comment (sven1977) [inline rllib/utils/tf_run_builder.py:85]:
```suggestion
                    "you have used `ray.init(local_mode=True)`. RLlib will use "
                    "timeline without `options=tf.RunOptions.FULL_TRACE`."
```

**Pull request merged**